ks3-standard/src/commonMain/kotlin/io/ks3/standard/SortedCollectionSerializer.kt

```diff
@@ -0,0 +1,31 @@
+package io.ks3.standard
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.serializer
+
+/**
+ * A serializer that sorts a collection with the given [comparator] upon serialization.
+ *
+ * Example:
+ * ```kotlin
+ * class SortedStringCollectionSerializer :
+ *    KSerializer<Collection<String>> by sortedCollectionSerializer(compareBy { it })
+ *
+ * @Serializable
+ * data class Friends(
+ *    @Serializable(SortedStringCollectionSerializer::class)
+ *    @Suppress("SERIALIZER_TYPE_INCOMPATIBLE")
+ *    val names: List<String>
+ * )
+ * ```
+ */
+inline fun <reified T> sortedCollectionSerializer(comparator: Comparator<in T>): KSerializer<Collection<T>> {
```

Comment:
WDYT about adding specific ones for Set and List? Would that remove the `SERIALIZER_TYPE_INCOMPATIBLE` warnings?

Reply:
I think duplicating is fine in this case 👍🏼